File: 백준/Bronze/2231. 분해합/분해합.py
N = int(input())
# N = 101x + 11y + 2z
for i in range(1, N + 1):
    num = list(map(int, str(i)))
    res = sum(num) + i
    if res == N:
        print(i)
        break
    if i == N:
        print(0)

File: 백준/Bronze/2231. 분해합/분해합.cc
#include <iostream>

using namespace std;

int main(){
    int N;
    cin >> N; 
    
    int res = 0;
    for(int i = 1; i < N; i++){
        int sum = 0;
        int num = i;
        while(true){
            if(num == 0)
                break;
            sum += num % 10;
            num /= 10;
        }
        
        if(sum + i == N){
            res = i;
            break;
        }
    }
    cout << res;
}